osquery/tables/system/windows/background_activities_moderator.cpp

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <osquery/core/core.h>
+#include <osquery/core/tables.h>
+#include <osquery/logger/logger.h>
+#include <osquery/tables/system/windows/registry.h>
+#include <osquery/utils/conversions/windows/windows_time.h>
+
+#include <string>
+
+namespace osquery {
+namespace tables {
+
+constexpr auto kBamRegPath =
+    "HKEY_LOCAL_MACHINE\\SYSTEM\\CurrentControlSet\\Services\\bam\\%%\\%%";
+
+QueryData genBackgroundActivitiesModerator(QueryContext& context) {
+  QueryData results;
+
+  std::set<std::string> bam_keys;
+  Row r;
+
+  expandRegistryGlobs(kBamRegPath, bam_keys);
+
+  for (const auto& rKey : bam_keys) {
+    std::size_t bam_entry = rKey.find("UserSettings\\S");
+    if (bam_entry == std::string::npos) {
+      continue;
+    }
+    std::size_t key = rKey.find("S-1");
+    if (key == std::string::npos) {
+      continue;
+    }
+    std::string sid = rKey.substr(key);
+
+    QueryData bam_entries;
+    queryKey(rKey, bam_entries);
+
+    for (const auto& bKey : bam_entries) {
+      r["path"] = bKey.at("name");
```

Comment:
Can you move the instantiation of `Row r` here?

This way you protect yourself from having any "left over" fields set in `r`.

Generally, prefer to allocate as close to the first usage as possible.

Reply:
understood, moved

Reply:
As commented, the `r` instantiation should be inside the forloop.

Reply:
whoops, misunderstood/misread
moved again XD